cmd/ontap-cvo.go

```diff
@@ -234,6 +236,68 @@ func checkExpired(labelValue string) bool {
 	return false
 }
 
+/*
+This will check for the existence of an S3 user
+https://docs.netapp.com/us-en/ontap-restapi/ontap/get-protocols-s3-services-users-.html
+Requires: svm.uuid, name, password and username for authentication
+Returns true if it does exist
+*/
+func checkIfS3UserExists(managementIP string, svmUuid string, onPremName string, username string, password string) bool {
+	// Build the request
+	//Encode the data
+	urlString := "https://" + managementIP + "/api/protocols/s3/services/" + svmUuid + "/users/" + onPremName
+	statusCode, _ := performHttpGet(username, password, urlString)
+	// if its 200
+	if statusCode == 200 {
+		return true
+	}
+	return false
+}
+
+/*
+Provides basic authentication
+*/
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.StdEncoding.EncodeToString([]byte(auth))
+}
+
+/*
+Does basic get for requests to the API. Returns the code and a json formatted response
+https://www.makeuseof.com/go-make-http-requests/
+apiPath should probably be /apiPath/
+*/
+func performHttpGet(username string, password string, url string) (statusCode int, responseBody []byte) {
+	// url := "https://" + managementIP + apiPath + filerUUID + "/users"
+	req, err := http.NewRequest("GET", url, nil)
+	req.Header.Set("Content-Type", "application/json")
+	// req.Header. // need to set other information
+	authorization := basicAuth(username, password) // this must be confirmed)
```

Comment:
"this must be confirmed", is that accurate?

Reply:
ah this was a copy paste error, this was in regards to the credentials for the account that Roham would give us to authenticate with ontap. Will be deleted